Maple2.Server.Game/PacketHandlers/ItemUseHandler.cs

```diff
@@ -38,4 +46,46 @@ public class ItemUseHandler : PacketHandler<GameSession> {
         
         session.Send(StoryBookPacket.Load(storyBookId));
     }
+
+    private static void HandleChatSticker(GameSession session, Item item) {
+        Dictionary<string, string> parameters = XmlParseUtil.GetParameters(item.Metadata?.Function?.Parameters);
+
+        if (!parameters.ContainsKey("id")) {
+            session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed));
+            return;
+        }
+
+        int stickerSetId = int.Parse(parameters["id"]);
+
+        int duration = 0;
+        if (parameters.TryGetValue(parameters["id"], out string? durationString)) {
+            int.TryParse(durationString, out duration);
+        }
+        
+        if (session.Player.Value.Unlock.StickerSets.ContainsKey(stickerSetId)) {
+            if (session.Player.Value.Unlock.StickerSets[stickerSetId] == long.MaxValue) {
+                session.Send(ChatStickerPacket.Error(ChatStickerError.s_msg_chat_emoticon_add_failed_already_exist));
+                return;
+            }
+
+            // does not expire
+            if (duration == 0) {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = long.MaxValue;
+            } else {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = Math.Min(session.Player.Value.Unlock.StickerSets[stickerSetId] + duration, long.MaxValue);
+            }
+        } else {
+            if (duration == 0) {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = long.MaxValue;
+            } else {
+                session.Player.Value.Unlock.StickerSets[stickerSetId] = duration + DateTime.Now.ToEpochSeconds();
```

Comment:
I use `DateTimeOffset.UtcNow.ToUnixTimeSeconds()` everywhere since `DateTime.Now` uses system timezone. There are exceptions when interacting with db, but it's complicated.